File: src/main/java/com/openpms/server/AppService.java
package com.openpms.server;

import cn.dev33.satoken.secure.BCrypt;
import cn.dev33.satoken.stp.StpUtil;
import com.openpms.server.dto.LoginParam;
import com.openpms.server.users.User;
import com.openpms.server.users.UserMapper;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;


@RequiredArgsConstructor
@Service
public class AppService {
    private final UserMapper userMapper;

    public Boolean login(LoginParam param) {
        User user = userMapper.findByEmail(param.getEmail());
        if (user == null) {
            return false;
        }
        if (!BCrypt.checkpw(param.getPassword(), user.getPassword())) {
            return false;
        }
        StpUtil.login(param.getEmail());
        return true;
    }
}

File: src/main/kotlin/com/creator/server/app/AppService.kt
package com.creator.server.app

import cn.dev33.satoken.secure.BCrypt
import cn.dev33.satoken.stp.StpUtil
import com.baomidou.mybatisplus.core.conditions.query.QueryWrapper
import com.creator.server.app.dto.*
import com.creator.server.user.User
import com.creator.server.user.UserMapper
import org.springframework.stereotype.Service

@Service
class AppService(
    private val userMapper: UserMapper
) {
    fun login(body: LoginBody): Boolean {
        val query = QueryWrapper<User>()
        query.eq("email", body.email)
            .eq("status", true)
        val user = userMapper.selectOne(query) ?: return false
        if (!BCrypt.checkpw(body.password, user.password)) {
            return false
        }
        StpUtil.login(body.email, false)
        return true
    }
}
